src/types/array.js

```diff
@@ -41,6 +41,18 @@ export default parameterized(T => class ArrayType {
     return [value, ...valueOf(this)];
   }
 
+  sort(compareFn) {
+    let init = valueOf(this);
+    let result = [...init].sort(compareFn);
+    let equalityCheck = (list1, list2) =>
```

Comment:
Your equalityCheck is a fully general equality check, but there are several constraints here that let us optimize.

1) This is a sort, so the resulting list will always have the same number of members as the list it was derived from. As such, the length check isn't necessary.
2) If you're going to define this function inline, I'd just allow it to reference the `init` closure variable which is implicit. Otherwise, you would want to define the function outside of the method.
3) I'd recommend either a `for` loop or `Array.prototype.every` which allow you to return early because you know that two lists are not equal  _the first_ time you experience an inequality. With the reduction here, you're going to iterate through the entire list even if you see that the first two items are different:


Here's the `for` loop version:

```js
function didChange(result) {
  for (let i = 0; i < init.length; i++) {
    if (init[i] !== result[i]) {
      return false;
    }
  }
  return true;
}
```
See how it fails fast? But we should probably go with [some](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some) or [every](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/every) since they'd be more concise and functionally "hip"

Reply:
👍 I've switched to using `Array.prototype.every` here instead.